taskcluster/android_taskgraph/transforms/build_aab.py

```diff
@@ -0,0 +1,129 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Apply some defaults and minor modifications to the jobs defined in the
+build-bundle (build AAB) kind.
+"""
+
+from android_taskgraph.build_config import get_variant
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import resolve_keyed_by
+
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def add_variant_config(config, tasks):
+    for task in tasks:
+        attributes = task.setdefault("attributes", {})
+        if not attributes.get("build-type"):
+            attributes["build-type"] = task["name"]
+        yield task
+
+
+@transforms.add
+def resolve_keys(config, tasks):
+    for task in tasks:
+        for field in ("optimization",):
+            resolve_keyed_by(
+                task,
+                field,
+                item_name=task["name"],
+                **{
+                    "tasks-for": config.params["tasks_for"],
+                },
+            )
+
+        yield task
+
+
+@transforms.add
+def build_pre_gradle_command(config, tasks):
+    for task in tasks:
+        source_project_name = task["source-project-name"]
+        pre_gradlew = task["run"].setdefault("pre-gradlew", [])
```

Comment:
Improvement for a follow-up: I love the stats you got in [bug 1567912 comment 7](https://bugzilla.mozilla.org/show_bug.cgi?id=1567912#c7). I think that's going to be useful to debug any bug around the size of the generated APK. We could add the call to `bundletool` in the `post-gradlew` step.

Reply:
Thanks, I like this idea...except, I would need to install `bundletool`, and its official home is a github repo (and with no "latest" release!); I wonder if adding that kind of external dependency is justified.

Reply:
Yes, of course - I somehow forgot about toolchains! I filed bug 1847221 for follow-up.